scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala

```diff
@@ -10,45 +12,47 @@ import org.scalatest._
+ * def apply(x: Int): Int = x * 2
+ * def unapply(z: Int): Option[Int] = if (z%2 == 0) Some(z/2) else None
  * }
  *
  * object TwiceTest extends Application {
- *   val x = Twice(21)
- *   x match { case Twice(n) => Console.println(n) } // prints 21
+ * val x = Twice(21)
+ * x match { case Twice(n) => Console.println(n) } // prints 21
  * }
  *
  *
  * There are two syntactic conventions at work here:
  *
- *  * The pattern `case Twice(n)` will cause an invocation of `Twice.unapply`, which is used to match even number; the return value of the `unapply` signals whether the argument has matched or not, and any sub-values that can be used for further matching. Here, the sub-value is `z/2`
- *  * The `apply` method is not necessary for pattern matching. It is only used to mimick a constructor. `val x = Twice(21)` expands to `val x = Twice.apply(21)`.
+ * * The pattern `case Twice(n)` will cause an invocation of `Twice.unapply`, which is used to match even number; the return value of the `unapply` signals whether the argument has matched or not, and any sub-values that can be used for further matching. Here, the sub-value is `z/2`
+ * * The `apply` method is not necessary for pattern matching. It is only used to mimick a constructor. `val x = Twice(21)` expands to `val x = Twice.apply(21)`.
  *
  * The code in the preceding example would be expanded as follows:
  *
  *
  * object TwiceTest extends Application {
- *   val x = Twice.apply(21)
- *   Twice.unapply(x) match { case Some(n) => Console.println(n) } // prints 21
+ * val x = Twice.apply(21)
+ * Twice.unapply(x) match { case Some(n) => Console.println(n) } // prints 21
  * }
  *
  * The return type of an `unapply` should be chosen as follows:
  *
- *  * If it is just a test, return a `Boolean`. For instance `case even()`
- *  * If it returns a single sub-value of type `T`, return a `Option[T]`
- *  * If you want to return several sub-values `T1,...,Tn`, group them in an optional tuple `Option[(T1,...,Tn)]`.
+ * * If it is just a test, return a `Boolean`. For instance `case even()`
+ * * If it returns a single sub-value of type `T`, return a `Option[T]`
+ * * If you want to return several sub-values `T1,...,Tn`, group them in an optional tuple `Option[(T1,...,Tn)]`.
  *
  * Sometimes, the number of sub-values is fixed and we would like to return a sequence. For this reason, you can also define patterns through `unapplySeq`. The last sub-value type `Tn` has to be `Seq[S]`. This mechanism is used for instance in pattern `case List(x1, ..., xn)`.
  *
  */
-class Extractors extends FlatSpec with Matchers {
+class Extractors extends FlatSpec with Matchers with Exercises {
 
 
   /**
    * When you create a case class, it automatically can be used with pattern matching since it has an extractor:
    */
-  def extractorCaseClasses(solutions: Seq[Any]): Unit = {
+  def caseClasses(res0: String): ExerciseResult[Unit] = ExerciseRunner("Case Classes") {
+
+    println("Inside case classes args : " + res0)
```

Comment:
This is actually just a sample class and should be replaced with the rest of the exercises in the stdlib package. I believe Rafa is porting all the Json data from v1 to this format and that point this class will dissappear